File: frontend/src/types/files.ts
export interface FileSummary {
  id: number
  filename: string
  status: string
  status_reason: string | null
  book_title: string | null
  book_author: string | null
  book_series: string | null
  book_series_number: number | null
  computed_confidence: number | null
  ai_reasoning: string | null
  quality_score: number | null
  discovered_at: string
}

File: backend/app/schemas/files.py
from pydantic import BaseModel


class ConfirmBatchRequest(BaseModel):
    file_ids: list[int]


class ConfirmBatchResult(BaseModel):
    confirmed: int
    skipped: int


class FileSummary(BaseModel):
    id: int
    filename: str
    status: str
    status_reason: str | None
    book_title: str | None
    book_author: str | None
    book_series: str | None
    book_series_number: float | None
    computed_confidence: int | None
    ai_reasoning: str | None
    quality_score: int | None
    discovered_at: str
